pagopa/io-app
# [IOBP-396] Add new `BonusCard` component into ID Pay initiative details screen

## Short description
This PR implements the new BonusCard component into ID Pay initiative details screen.
This PR also introduces small refactoring to ID Pay initiative details screen (name and exports).

## List of changes proposed in this pull request
- Renamed `InitiativeDetailsScreen` to `IdPayInitiativeDetailsScreen`
- Renamed `BeneficiaryDetailsScreen` to `IdPayBeneficiaryDetailsScreen`
- Renamed `OperationsListScreen` to `IdPayOperationsListScreen`
- Added `BonusCardScreenComponent` into `IdPayInitiativeDetailsScreen`
- Removed deprecated components (`InitiativeStatusLabel`, `InitiativeDetailsBaseScreenComponent`)
- Fixed tests
- Added second level header to `IdPayBeneficiaryDetailsScreen` and `IdPayOperationsListScreen`

## How to test
Navigate to the details screen of an ID Pay initiative. Check that everything is displayed correctly, with and without the new design system feature flag enable.

## Previews

| iOS | Android |
| --- | --- |
| <video src="https://github.com/pagopa/io-app/assets/6160324/ad0cddb1-e97a-4c41-b139-f56b76e4890a" width="200" />| <video src="https://github.com/pagopa/io-app/assets/6160324/eea1cdaa-a800-4f3b-9dbd-e58c3bbbabb0" width="200" />|
